pages/blog/tag/[tag].js

```diff
@@ -46,22 +37,23 @@ export async function getStaticPaths() {
 
 export async function getStaticProps({ params, preview = false }) {
   const posts = await ContentfulApi.getAllBlogPosts();
-  
-  console.log(posts);
 
-  posts = posts[0].contentfulMetadata.tags[0].id.filter(params.tag);
+  const relatedPosts = posts.reduce((acc, post) => {
+    if (
+      post.contentfulMetadata &&
+      post.contentfulMetadata.tags &&
+      post.contentfulMetadata.tags.find(({ id }) => id === params.tag)
```

Comment:
@jazzybuilds this is just a really extensive guard against properties not existing on the object but at the end we look to see if the current `tag` exists in the list of `tags` on the current `post` we are looping on. We then push that `post` to the accumulator and we have the related posts for that `tag`.